### ISSUE TITLE
Media query parentheses space causes and media-feature-no-missing-punctuation

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#742]
These three following use cases should be valid for the rule `media-feature-no-missing-punctuation`

Currently only the 1st example will pass, the 2nd and 3rd fails with `Unexpected missing punctuation within non-boolean media feature (media-feature-no-missing-punctuation)`

1 -  If `media-query-parentheses-space-inside` is `never`: 

```css
@media all and (max-width: 699px) and (min-width: 520px) {

	/* Your selectors */
}
```

2 -  If `media-query-parentheses-space-inside` is `always`:
```css
@media all and ( max-width: 699px ) and ( min-width: 520px ) {

	/* Your selectors */
}
```

3 - If `media-query-parentheses-space-inside` is not defined:
```css
@media all and (max-width: 699px) and (min-width: 520px) {

	/* Your selectors */
}

@media all and ( max-width: 699px ) and ( min-width: 520px ) {

	/* Your selectors */
}
```

**Comment by davidtheclark** [comment]
Thanks for finding the bug, @ntwb, and for the failing tests (very helpful).

**Comment by KeironLowe92** [comment]
This error still occurs when using nested brackets for calculations, such as:

```
@media (min-width: 45em) and (max-width: (1000px + $inuit-base-spacing-unit--large)) {
    display: none;
}
```

Still fails if I replace the variable with a number, passes if I remove the spaces around the `+`.

**Comment by evilebottnawi** [comment]
@KeironLowe92 be good if you PR tests failed

**Comment by KeironLowe92** [comment]
@evilebottnawi Happy to submit one, got any examples of PR tests? Not sure how they should be done

**Comment by evilebottnawi** [comment]
@KeironLowe92 https://github.com/stylelint/stylelint/pull/1047
